### PR TITLE
Clarify unit of measurement warning for price entity

### PR DESCRIPTION
**Pull request by bramkragten**
## Proposed change

image
## Type of change

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (thank you!)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [x] Code quality improvements to existing code or addition of tests

## Example configuration

```yaml

```

## Additional information

- This PR fixes or closes issue: fixes #
- This PR is related to issue or discussion:
- Link to documentation pull request:

## Checklist

- [ ] The code change is tested and works locally.
- [ ] There is no commented out code in this PR.
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

[docs-repository]: https://github.com/home-assistant/home-assistant.io

### CODE REVIEW

File: src/translations/en.json [hunk 1]
```diff
@@ -1107,15 +1107,15 @@
               },
               "entity_unexpected_unit_energy": {
                 "title": "Unexpected unit of measurement",
-                "description": "The following entities do not have expected units of measurement kWh or Wh:"
+                "description": "The following entities do not have the expected units of measurement ''kWh'' or ''Wh'':"
```
Comment (ludeeus) [inline src/translations/en.json:1110]: We generally use `\"`  to add and escape quotes

File: src/panels/config/energy/components/ha-energy-validation-result.ts [hunk 2]
```diff
@@ -31,7 +31,10 @@ class EnergyValidationMessage extends LitElement {
             }
           >
             ${this.hass.localize(
-              `ui.panel.config.energy.validation.issues.${issueType}.description`
+              `ui.panel.config.energy.validation.issues.${issueType}.description`,
+              issueType === "entity_unexpected_unit_price"
+                ? { currency: this.hass.config.currency }
```
Comment (ludeeus) [inline src/panels/config/energy/components/ha-energy-validation-result.ts:36]: Should we use a "currency" placeholder here?
`expected units of measurement "/kWh" or "/Wh"` is misleading as well?

Comment (bramkragten) [reply src/panels/config/energy/components/ha-energy-validation-result.ts:36]: The currency will always be defined?

Comment (ludeeus) [reply src/panels/config/energy/components/ha-energy-validation-result.ts:36]: Right!👍

File: src/translations/en.json (hunk 1 above)
Comment (bramkragten) [reply src/translations/en.json:1110]: We seem to use `'` quite a lot, a bit 50/50... I personally like `'` better, but also 🤷‍♂️

**Review by ludeeus — APPROVED** [review]

**Pull request merged**